File: 01_Language/01_Functions/python/file_put_contents.py
# coding: utf-8

def file_put_contents(filename, data, append=False):
    mode = "a" if append else "w"
    with open(filename, mode=mode) as f:
        f.write(data)


if __name__ == '__main__':
    from file_get_contents import file_get_contents

    file_put_contents("test.txt", "John Smith\n", True)
    print(file_get_contents("test.txt"))

File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/FilePutContents.java
package com.onevgo.functions;

import com.google.common.io.FileWriteMode;
import com.google.common.io.Files;

import java.io.File;
import java.io.IOException;

public class FilePutContents {
    public static int filePutContents(String filename, byte[] data, FileWriteMode... flags) {
        try {
            Files.asByteSink(new File(filename), flags).write(data);
            return data.length;
        } catch (IOException e) {
            e.printStackTrace();
        }
        return -1;
    }

    public static int filePutContents(String filename, String data, FileWriteMode... flags) {
        return filePutContents(filename, data.getBytes(), flags);
    }

    public static void main(String[] args) {
        filePutContents("test.txt", "John Smith\n".getBytes(), FileWriteMode.APPEND);
        System.out.println(FileGetContents.fileGetContents("test.txt"));
    }
}
